primitives/benches/crypto_primitives/poseidon_crh.rs

```diff
@@ -0,0 +1,61 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use algebra::{
+    fields::{
+        mnt4753::Fr as MNT4753Fr,
+        mnt6753::Fr as MNT6753Fr,
+    }
+};
+
+use algebra::UniformRand;
+use rand_xorshift::XorShiftRng;
+use rand::SeedableRng;
+use primitives::crh::{
+    poseidon::{
+        parameters::{MNT4753PoseidonParameters, MNT6753PoseidonParameters},
+        PoseidonHash
+    },
+    FieldBasedHash,
+};
+
+type MNT4PoseidonHash = PoseidonHash<MNT4753Fr, MNT4753PoseidonParameters>;
+type MNT6PoseidonHash = PoseidonHash<MNT6753Fr, MNT6753PoseidonParameters>;
+
+fn poseidon_crh_eval_mnt4(c: &mut Criterion) {
+
+    let mut rng = XorShiftRng::seed_from_u64(1231275789u64);
+    let mut h = MNT4PoseidonHash::init(None);
+
+    c.bench_function("Poseidon CRH Eval for MNT4", move |b| {
+        b.iter(|| {
+            for _ in 0..2000 {
+                h.update(MNT4753Fr::rand(&mut rng));
+            }
+            h.finalize();
+        })
+    });
+}
+
+fn poseidon_crh_eval_mnt6(c: &mut Criterion) {
+
+    let mut rng = XorShiftRng::seed_from_u64(1231275789u64);
+    let mut h = MNT6PoseidonHash::init(None);
+
+    c.bench_function("Poseidon CRH Eval for MNT6", move |b| {
+        b.iter(|| {
+            for _ in 0..2000 {
```

Comment:
Again, here. Why 2000 hashes?

Reply:
I mean, actually they are 2000 inputs. Since our rate is 2, this means 1000 hashes no ?

Reply:
Same as above.